src/assets/linkPreview.js

```diff
@@ -0,0 +1,250 @@
+  }
+};
+
+const linkPreviewPopup = () => {
+  const { origin } = new URL(window.location);
+  const fetchLinkPreview = async (elementHref, popupContainer) => {
+    try {
+      const linkPreviewRes = await fetch(`${origin}/pdftron-link-preview?url=${elementHref}`);
+      const { hostname } = new URL(elementHref);
+      if (linkPreviewRes.status !== 400) {
+        const linkPreviewResJson = await linkPreviewRes.json();
+        const { faviconUrl, pageTitle, metaDescription } = linkPreviewResJson;
+        const faviconDiv = faviconUrl ? `${faviconEmptyString}<img class="pdftron-link-favicon" width="20" src="${faviconUrl}" onload="successLoadingImage(this);">` : faviconEmptyString;
+        const hostNameDiv = pageTitle ? `<div style="color: #868E96 !important; margin-top: 4px;">${getHostName(hostname)}</div>` : '';
+        const metaDiv = metaDescription ? `<div class="pdftron-link-meta">${metaDescription}</div>` : '';
+
+        popupContainer.setAttribute('data-pdftronpreview', 'pdftron-link-fullpreview');
+        popupContainer.innerHTML = `
+          <div class="pdftron-link-popup-inner">
+            ${popupDefaultInnerHTML(faviconDiv, pageTitle || elementHref, elementHref)}
+            <div style="margin-left: 24px;">
+              ${hostNameDiv}
+              ${metaDiv}
+            </div>
+          </div>
+        `;
+
+        // by default, clicking on the popup will be the same as clicking on the <a>
+        popupContainer.onclick = ((e) => e.preventDefault());
+      }
+    } catch (err) {
+      console.error('Link preview', elementHref, err);
+    }
+  };
+
+  // block navigation for suspicious <a> that don't have href or empty href: stubbing onclick
+  // block navigation for all a tags that don't start with #
+  document.querySelectorAll(linkSelectors).forEach((elem) => {
+    // after href has already been modified
+    const elementHref = elem.getAttribute('href');
+
+    if (isURLAbsolute(elem.href) && elem.hasChildNodes()) {
+      const elChildNodes = Array.from(elem.childNodes);
+      // check if childNodes has some that is an element and doesn't have data-pdftron
+      if (!elChildNodes.some((childEL) => childEL.nodeType === Node.ELEMENT_NODE && childEL.dataset.pdftron === 'pdftron-link-popup')) {
+        const popupContainer = document.createElement('div');
+        popupContainer.setAttribute('data-pdftron', 'pdftron-link-popup');
+        popupContainer.className = 'pdftron-link-popup-outer';
+        popupContainer.innerHTML = `
+          <div class="pdftron-link-popup-inner">
+            ${popupDefaultInnerHTML(faviconEmptyString, elementHref, elementHref)}
+          </div>
+        `;
+
+        elem.appendChild(popupContainer);
+        let undoOverflow;
+        let intersectionobserver;
+        let elemOriginalStyles = [];
+        let undoFunctions = [];
+
+        const traverseToParentWithOverflowHidden = (targetElement) => {
+          const parentElement = targetElement?.parentElement;
```

Comment:
This name doesn't seem accurate. Wouldn't `setParentsToOverflowVisible` be more accurate?